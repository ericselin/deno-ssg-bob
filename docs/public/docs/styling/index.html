<html lang="en"><head><meta charSet="UTF-8"></meta><meta name="viewport" content="width=device-width, initial-scale=1.0"></meta><meta http-equiv="X-UA-Compatible" content="ie=edge"></meta><title>bob the static site builder</title><link rel="preconnect" href="https://fonts.gstatic.com"></link><link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Code+Pro:wght@400;700;900&display=swap" rel="stylesheet"></link><style>:root {
  --nord0: #2e3440;
  --nord1: #3b4252;
  --nord2: #434c5e;
  --nord3: #4c566a;
  --nord4: #d8dee9;
  --nord5: #e5e9f0;
  --nord6: #eceff4;
  --nord7: #8fbcbb;
  --nord8: #88c0d0;
  --nord9: #81a1c1;
  --nord10: #5e81ac;
  --nord11: #bf616a;
  --nord12: #d08770;
  --nord13: #ebcb8b;
  --nord14: #a3be8c;
  --nord15: #b48ead;

  --margin: 1rem;
  --max-width: min(600px, 100% - 4 * var(--margin));

  --color-bg: var(--nord0);
  --color-fg: var(--nord6);
  --color-body: var(--nord1);
  --color-indent: var(--nord3);
  --color-muted: var(--nord4);
  --color-accent: var(--nord8);

  --font-text: 'Inter', sans-serif;
  --font-heading: 'Source Code Pro', monospace;
  --font-mono: 'Source Code Pro', monospace;
}

@media (min-width: 500px) {
  :root {
    --margin: 1.5rem;
  }
}

@media (min-width: 768px) {
  :root {
    --margin: 2rem;
  }
}

body {
  padding: 0;
  margin: 0;
  font-family: var(--font-text);
  background: var(--color-body);
  color: var(--color-fg);
}

body > header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: var(--margin);
  color: var(--color-muted);
  font-family: var(--font-heading);
}

body > header > a {
  text-decoration: none;
  font-weight: 900;
  font-size: 1.5rem;
  background: var(--color-accent);
  text-align: center;
  border-radius: 25%;
  color: var(--color-body);
  padding: 0 0.25rem;
}

body > header > a:hover {
  color: var(--color-bg);
}

body > header ul {
  display: grid;
  grid-auto-flow: column;
  gap: 1rem;
  list-style: none;
  padding: 0;
  margin: 0;
}

body > h1 {
  max-width: var(--max-width);
  margin: calc(2 * var(--margin)) auto;
  padding: 0 var(--margin);
  color: var(--color-fg);
  font-family: var(--font-heading);
}

main {
  max-width: var(--max-width);
  margin: var(--margin) auto;
  padding: 0 var(--margin);
  background: var(--color-bg);
  border-radius: 1rem;
  line-height: 1.25;
}

main::before,
main::after {
  content: ' ';
  display: block;
  contain: layout;
  margin: var(--margin) 0;
}

h1 a,
h2 a,
h3 a,
h4 a,
h5 a,
h6 a {
  text-decoration: none;
}

a {
  color: inherit;
  transition: color ease-in 200ms;
}

a:hover {
  color: var(--color-accent);
}

pre {
  max-width: 100%;
  display: block;
  overflow: auto;
}

code {
  font-family: var(--font-mono);
  line-height: initial;
}

article + article {
  margin-top: calc(2 * var(--margin));
}

article + article::before {
  content: '';
  display: block;
  height: 1px;
  width: 50%;
  background: var(--color-indent);
  margin: 0 auto calc(2 * var(--margin));
}

img {
  display: block;
  max-width: 100%;
}

.button-link {
  border: 1px solid var(--color-accent);
  border-radius: 0.5rem;
  padding: 0.5rem 1rem;
  display: inline-block;
  text-decoration: none;
}

body > footer {
  margin: var(--margin);
  text-align: center;
  color: var(--color-muted);
  font-family: var(--font-heading);
}

body > nav ul {
  display: grid;
  grid-auto-flow: column;
  list-style: none;
  padding: 0;
  margin: var(--margin);
  justify-content: center;
  gap: 1rem;
}
</style></head><body><header><a href="/">bob</a><ul><li><a href="/getting-started/">Getting started</a></li><li><a href="/scenarios/">Common scenarios</a></li></ul></header><nav><ul><li><a href="/docs/styling/">Styling and CSS</a></li><li><a href="/docs/">Getting started</a></li></ul></nav><main><h1>Styling and CSS</h1><p>Style your component with a good ol&#39; CSS file and mark it as a dependency to your component. Then, render the needed (and only the needed) CSS directly into the HTML <code>&lt;head&gt;</code>.</p>

<ol>
<li>Add the layout-directory-relative path to the CSS file in the <code>needsCss</code> property of your <code>Component</code>.</li>
<li>In your base layout (or wherever you render into the <code>&lt;head&gt;</code>, use the <code>Contenxt.neededCss</code> property to get a list of needed CSS filenames.</li>
<li>Render the content of the CSS files into the head using the <code>readContents()</code> helper.</li>
</ol>
</main><footer>Static sites should be fast, including updating content</footer></body></html>