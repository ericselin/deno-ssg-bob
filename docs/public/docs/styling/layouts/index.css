h1 {
  color: blue;
}

main {
  margin: 1rem;
  padding: 0 1rem;
  border: 1px solid green;
}
