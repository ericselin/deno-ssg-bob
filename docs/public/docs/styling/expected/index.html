<html lang="en"><head><title>This is my page</title><style>h1 {
  color: blue;
}

main {
  margin: 1rem;
  padding: 0 1rem;
  border: 1px solid green;
}

body {
  font-family: serif;
  margin: 1rem auto;
  max-width: 50rem;
}
</style></head><body><main><h1 id="hello">Hello</h1>
<p>This here is my page!</p>
</main></body></html>